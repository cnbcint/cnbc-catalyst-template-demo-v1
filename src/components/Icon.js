import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
const Container = styled.span`
  .ion-icon {
    color: ${(props) => props.iconColor};
  }
`;
const Variants = { filled: "", outline: "-outline", sharp: "-sharp" };
/**
 * Wrapper for Ionic Icons more info at https://ionicons.com/
 */
export default function Icon(props) {
  return (
    <Container iconColor={props.color}>
      <ion-icon
        class="ion-icon"
        name={props.name + Variants[props.variant]}
        size={props.size}
      ></ion-icon>
    </Container>
  );
}
Icon.defaultProps = {
  variant: "filled",
  size: "small",
  color: "inherit",
};

Icon.propTypes = {
  /**
   * Icon Name see https://ionicons.com/ for details
   */
  name: PropTypes.string.isRequired,
  /**
   * Icon type
   */
  variant: PropTypes.oneOf(["filled", "outline", "sharp"]),
  /**
   * Icon size
   */
  size: PropTypes.oneOf(["small", "large"]),
  /**
   * Icon color
   */
  color: PropTypes.string,
};
