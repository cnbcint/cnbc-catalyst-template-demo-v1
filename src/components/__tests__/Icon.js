/* eslint react/display-name: 0 */
import React from "react";
import Icon from "../Icon";
import { render } from "@testing-library/react";
import "jest-styled-components";

describe("Icon", () => {
  it("Renders correctly", () => {
    const { container } = render(<Icon name="heart" variant="outline" />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("Add the right color", () => {
    const { container } = render(
      <Icon name="heart" variant="outline" color="blue" />
    );
    expect(container.firstChild).toHaveStyleRule("color", "blue", {
      modifier: ".ion-icon ",
    });
  });
});
