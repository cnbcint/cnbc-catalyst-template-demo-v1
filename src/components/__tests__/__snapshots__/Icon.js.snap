// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Renders correctly 1`] = `
.c0 .ion-icon {
  color: inherit;
}

<span
  class="c0"
>
  <ion-icon
    class="ion-icon"
    name="heart-outline"
    size="small"
  />
</span>
`;
